var/spack/repos/builtin/packages/mpip/package.py

```diff
@@ -4,37 +4,127 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 from spack import *
-import os
 
 
 class Mpip(AutotoolsPackage):
+
     """mpiP: Lightweight, Scalable MPI Profiling"""
-    homepage = "http://mpip.sourceforge.net/"
-    url      = "http://downloads.sourceforge.net/project/mpip/mpiP/mpiP-3.4.1/mpiP-3.4.1.tar.gz"
+    homepage = "https://software.llnl.gov/mpiP/"
+    url      = "https://github.com/LLNL/mpiP/releases/download/3.5/mpip-3.5.tgz"
     git      = "https://github.com/llnl/mpip.git"
+    maintainers = ['cchambreau']
 
     version('master', branch='master')
-    version("3.4.1", sha256="688bf37d73211e6a915f9fc59c358282a266d166c0a10af07a38a01a473296f0")
+    version('3.5',
+            sha256="e366843d53fa016fb03903e51c8aac901aa5155edabe64698a8d6fa618a03bbd")
+    version("3.4.1",
+            sha256="66a86dafde61546be80a130c46e4295f47fb764cf312ae62c70a6dc456a59dac")
+
+    variant('demangling', default=True,
+            description="Build with demangling support")
+
+    variant('setjmp', default=False,
+            description="Use setjmp to generate stack trace")
+
+    variant('mpi_io', default=True,
+            description="Enable MPI-I/O reporting")
+
+    variant('mpi_rma', default=True,
+            description="Enable MPI RMA reporting")
+
+    variant('mpi_nbc', default=True,
+            description="Enable MPI non-blocking collective reporting")
+
+    variant('bfd', default=True,
+            description="Enable GNU binutils libbfd for source lookup")
+
+    variant('libunwind', default=True,
+            description="Use libunwind togenerate stack trace")
 
-    variant('shared', default=False, description="Build the shared library")
-    variant('demangling', default=False, description="Build with demangling support")
-    variant('setjmp',
-            default=False,
-            description="Replace glic backtrace() with setjmp for stack trace")
+    variant('maxargs', values=int, default=32,
+            description='Set number of command line arguments in report')
 
-    depends_on("elf")
-    depends_on("libdwarf")
-    depends_on('libunwind', when=os.uname()[4] == "x86_64")
-    depends_on("mpi")
+    variant('stackdepth', values=int, default=8,
+            description='Specify maximum report stacktrace depth')
+
+    variant('internal_stackdepth', values=int, default=3,
+            description='Specify number of internal stack frames')
+
+    variant('add_shared_target', default=False, description="Add shared make target")
+
+    conflicts('platform=darwin')
+
+    # make-wrappers.py wrapper generator script requres python
```

Comment:
```suggestion
    # make-wrappers.py wrapper generator script requires python
```
Is this at build-time or at run-time? It sounds like you don't link to the Python libraries, so the current deptype is incorrect.